src/lib/core/processes/processManager.js

```diff
@@ -21,21 +26,21 @@ class ProcessManager {
     const self = this;
     self.plugin = this.plugins.createPlugin('processManager', {});
 
-    this.servicesState = {};
-    this.events.on("servicesState", (servicesState) => {
-      this.servicesState = servicesState;
-    });
-
     self.plugin.registerAPICall(
       'get',
       '/embark-api/services',
       (req, res) => {
-        let processList = [];
-        for (let serviceName in this.servicesState) {
-          let service = this.servicesState[serviceName];
-          processList.push({state: service.status, name: serviceName, description: service.name});
-        }
-        res.send(processList);
+        res.send(this._sevicesForApi(this.servicesState));
+      }
+    );
+
+    self.plugin.registerAPICall(
+      'ws',
+      '/embark-api/services',
+      (ws, _res) => {
+        this.events.on('servicesState', (servicesState) => {
+          ws.send(JSON.stringify(this._sevicesForApi(servicesState)), () => undefined);
```

Comment:
Curious why it's `undefined` here instead of `() => {}`?

Reply:
Because I feel it is more clear on the intention, when you do:
`() => {}` you are actually creating a function that return undefined all the time but it is implicit
if you do
`() => undefined`, you don't have to think about it and you immediately know that this will return undefined

So I prefer to be explicit.

also, there is an eslint rule that we don't have enabled that says: no empty function :)

 If you don't like it, or have strong opinion about it I am fine using the `() => {}` syntax

Reply:
Nope, I'm ok with either, it was more of a question as I'm used to seeing it the other way. Thanks :)